apps/prairielearn/src/api/trpc/routers/course-files/deleteQuestion.ts

```diff
@@ -0,0 +1,70 @@
+import { z } from 'zod';
+
+import { IdSchema } from '../../../../lib/db-types.js';
+import { QuestionDeleteEditor } from '../../../../lib/editors.js';
+import { selectCourseById } from '../../../../models/course.js';
+import { selectQuestionById } from '../../../../models/question.js';
+import { privateProcedure, selectUsers } from '../../trpc.js';
+
+export const deleteQuestion = privateProcedure
+  .input(
+    z.object({
+      // Context.
+      course_id: IdSchema,
+      user_id: IdSchema,
+      authn_user_id: IdSchema,
+      has_course_permission_edit: z.boolean(),
+
+      // Question data.
+      question_id: z.string(),
+    }),
+  )
+  .output(
+    z.union([
+      z.object({
+        status: z.literal('success'),
+        job_sequence_id: z.string(),
+      }),
+      z.object({
+        status: z.literal('error'),
+        job_sequence_id: z.string(),
+      }),
+    ]),
+  )
+  .mutation(async (opts) => {
+    const course = await selectCourseById(opts.input.course_id);
+    const { user, authn_user } = await selectUsers({
+      user_id: opts.input.user_id,
+      authn_user_id: opts.input.authn_user_id,
+    });
+
+    const question = await selectQuestionById(opts.input.question_id);
+
+    const editor = new QuestionDeleteEditor({
```

Comment:
leeching off of single deletion editor for now even though we only use this rpc for batch deletions- see discussion below and lmk if you want me to change it.

Reply:
I ended up removing bulk deletion for now in f55db8a. We can add it (and a broper batch-deletion endpoint) in another PR.